test/unit/helper/TypeMapping.js

```diff
@@ -0,0 +1,208 @@
+
+    t.equal(typeof TypeMapping.prototype.generateDynamicMappings, 'function', 'generateDynamicMappings() is a function');
+    
+    t.equal(typeof TypeMapping.prototype.loadTargets, 'function', 'loadTargets() is a function');
+    t.equal(typeof TypeMapping.prototype.load, 'function', 'load() is a function');
+
+    t.end();
+  });
+};
+
+module.exports.tests.constructor = function(test) {
+  test('constructor', function(t) {
+
+    var doc = new TypeMapping();
+
+    // initial values
+    t.deepEqual(doc.sources, [], 'initial value');
+    t.deepEqual(doc.source_aliases, {}, 'initial value');
+    t.deepEqual(doc.layers, [], 'initial value');
+    t.deepEqual(doc.layers_by_source, {}, 'initial value');
+    t.deepEqual(doc.layer_aliases, {}, 'initial value');
+    t.deepEqual(doc.source_mapping, {}, 'initial value');
+    t.deepEqual(doc.layer_mapping, {}, 'initial value');
+
+    t.end();
+  });
+};
+
+module.exports.tests.addStandardTargetsToAliases = function(test) {
+  test('static method addStandardTargetsToAliases', function(t) {
+
+    var aliases = { test: ['test2'] };
+
+    t.deepEqual(
+      TypeMapping.addStandardTargetsToAliases([], aliases),
+      { test: ['test2'] }
+    );
+    t.deepEqual(aliases, aliases, 'aliases object not mutated');
+
+    t.deepEqual(
+      TypeMapping.addStandardTargetsToAliases(['test'], aliases),
+      { test: ['test2'] },
+      'not modified'
+    );
+    t.deepEqual(aliases, aliases, 'aliases object not mutated');
+
+    t.deepEqual(
+      TypeMapping.addStandardTargetsToAliases(['baz'], aliases),
+      { test: ['test2'], baz: ['baz'] }
+    );
+    t.deepEqual(aliases, aliases, 'aliases object not mutated');
+
+    t.deepEqual(
+      TypeMapping.addStandardTargetsToAliases(['baz','boo'], aliases),
+      { test: ['test2'], baz: ['baz'], boo: ['boo'] }
+    );
+    t.deepEqual(aliases, aliases, 'aliases object not mutated');
+
+    t.end();
+
+  });
+};
+
+module.exports.tests.setSourceAliases = function(test) {
+  test('setter setSourceAliases', function(t) {
+    var tm = new TypeMapping();
+    t.deepEqual(tm.source_aliases, {});
+    tm.setSourceAliases({ foo: ['foo', 'bar'] });
+    t.deepEqual(tm.source_aliases, { foo: ['foo', 'bar'] });
+    t.end();
+  });
+};
+
+module.exports.tests.setLayersBySource = function(test) {
+  test('setter setLayersBySource', function(t) {
+    var tm = new TypeMapping();
+    t.deepEqual(tm.layers_by_source, {});
+    tm.setLayersBySource({ foo: ['foo', 'bar'] });
+    t.deepEqual(tm.layers_by_source, { foo: ['foo', 'bar'] });
+    t.end();
+  });
+};
+
+module.exports.tests.setLayerAliases = function(test) {
+  test('setter setLayerAliases', function(t) {
+    var tm = new TypeMapping();
+    t.deepEqual(tm.layer_aliases, {});
+    tm.setLayerAliases({ foo: ['foo', 'bar'] });
+    t.deepEqual(tm.layer_aliases, { foo: ['foo', 'bar'] });
+    t.end();
+  });
+};
+
+module.exports.tests.generateDynamicMappings = function(test) {
```

Comment:
This shadows an Elasticsearch term which means something pretty different, maybe we can come up with a different name?